### ISSUE TITLE
Install probleme on MacOS 14.4.1

### ISSUE THREAD

**Issue by ELEGER** [bruceravel/demeter#71]
Hello, 
I'm trying to install Demeter on a MacOS machine.
X11 version 1.20.11

After installing macport and X11,  I ran : "sudo port install xorg-server demeter"
and finished with

``
--->  Fetching archive for ifeffit

--->  Attempting to fetch ifeffit-1.2.13_8+gcc13.darwin_23.arm64.tbz2 from https://packages.macports.org/ifeffit

--->  Attempting to fetch ifeffit-1.2.13_8+gcc13.darwin_23.arm64.tbz2 from https://mse.uk.packages.macports.org/ifeffit

--->  Attempting to fetch ifeffit-1.2.13_8+gcc13.darwin_23.arm64.tbz2 from https://fra.de.packages.macports.org/ifeffit

--->  Building ifeffit

Error: Failed to build ifeffit: command execution failed
Error: See /opt/local/var/macports/logs/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_science_ifeffit/ifeffit/main.log for details.
Error: Follow https://guide.macports.org/#project.tickets if you believe there
is a bug.
Error: Processing of port demeter failed
``
Did anyone went through the same kind of issue ?

Thanks in advance

Emmanuel

**Comment by bruceravel** [comment]
Building all of demeter on a Mac is very challenging and is not something that I am able to support.  It would be much easier for you to install a virtual Windows machine and use the Windows installer package.

**Comment by ELEGER** [comment]
ok thanks.
Could you leave the issue open a little while ? Maybe/probably I'm not the only one with this issue and I am certain that someone has a solution :).
Cheers

**Comment by newville** [comment]
@ELEGER Well, it's not like "getting Demeter to work on a Mac is hard" is a new problem. See #60 or #64, or the documentation page at https://bruceravel.github.io/demeter/#mac.  Over at the Ifeffit mailing list, this seems to come up every 6 months or so. It seems that some people can sometimes get it to work, but I am not certain that there is a durable solution.  

It looks like the errors you are seeing are from building the ifeffit library.  Development and support for the ifeffit library ended 10 years ago.   Build problems will become inevitable, and I am not aware of anyone trying to fix or maintain those.   It is free software so someone can try to build and/or use it, but I sure wish they wouldn't.

**Comment by ELEGER** [comment]
Thanks for your answer @newville . 
I'll switch to linux/winos machine.
Best
Emmanuel

**Comment by bruceravel** [comment]
I think what Matt is trying to say is that you should take a look at Larch http://xraypy.github.io/xraylarch/

**Comment by newville** [comment]
@Yike-Ye   I would be curious to know how that installation process compares to 

```
> curl -O https://raw.githubusercontent.com/xraypy/xraylarch/master/installers/GetLarch.sh
> sh GetLarch.sh
```

If your demeter installation uses the ifeffit library, I will remind you that support for that library ended more than 10 years ago.  It has several bugs that will not be fixed.   Demeter can use Larch, but as the development and support of demeter waned, we also took on GUI development within the Larch codes, which continue to be developed and supported.  

The Larix GUI aims to be approximately feature-compatible with Athena/Artemis.  Some things are slightly different, but most things should be pretty familiar. There are some definite improvements in functionality due to using the Larch/lmfit/scipy libraries instead of ifeffit.  I would not call anything in the Larix GUI an improvement over Athena/Artemis.  The list of features in Athena/Artemis that are not in Larix is small.  We're happy to discuss and work on those. We have active development and discussions.  For a recent example, see https://github.com/xraypy/xraylarch/wiki/Development-Roadmap

**Comment by Yike-Ye** [comment]
Hi Matt @newville ,

To be honest, I didn't have any problems on installing ifeffit. Instead, the old version of wxPerl (built with wxWidgets 3.0) troubled me a lot. I managed to find an updated version of Wx-3.005 built from wxWidgets 3.2 on GitHub, and I used that to successfully installed Demeter. Things now look good till now. Since I'm a freshman to the software, I didn't test too many features.

Best,

Yike
